### ISSUE TITLE
WIP: Adding gui

### ISSUE THREAD

**Issue by sobkas** [Mord3rca/gamma-launcher#137]
I'm slowly working on GUI for launcher

**Comment by Sneethe** [comment]
If you filter people, by requiring terminal usage. I guarantee you will get less github issues being made for "How play game?" and "game not work"

**Comment by francislen99** [comment]
Any update on the GUI?

**Comment by dev-reimu** [comment]
@sobkas Any update on this?

**Comment by clockwinder** [comment]
@sobkas Any progress?  Any way we can contribute?

**Comment by sobkas** [comment]
Ok, first look at gui:
https://github.com/sobkas/gamma-launcher/commit/fb37478b3df604bc6eb4b0d7737957bf0debe7d3
Still work in progress.

**Comment by sobkas** [comment]
![Image](https://github.com/user-attachments/assets/a607f64d-9ca5-415e-84e7-459b341e10aa)

**Comment by sobkas** [comment]
I would appreciate some feedback

**Comment by Mord3rca** [comment]
@sobkas Lookin' good !

Keep in mind this GUI have to be optional :) btw which GUI framework are you using (wxWidget, Qt, ...) ?

**Comment by francislen99** [comment]
> I would appreciate some feedback

So far, it looks good. What I would personally appreciate a lot is if the launcher would be able to update the existing gamma folders once newer gamma versions get released.

**Comment by sobkas** [comment]
> [@sobkas](https://github.com/sobkas) Lookin' good !
> 
> Keep in mind this GUI have to be optional :) btw which GUI framework are you using (wxWidget, Qt, ...) ?

Yes I'm also thinking about making it optional, as gtk4 and family is rather big
Btw you run GUI with gui-gamma-launcher while CLI with gamma-launcher
Also gui-gamma-launcher works, it can install anomaly, do full-install and do usvfs-workaround

**Comment by francislen99** [comment]
@sobkas Not sure. If im not mistaken in the official GUI for gamma in windows it has a "update gamma" function where when there is a new version/fixes you can update into the latest gamma version without having to do a complete reinstall.

**Comment by sobkas** [comment]
> > Also gui-gamma-launcher works, it can install anomaly, do full-install and do usvfs-workaround
> 
> How are you doing that ? By forking and redirecting stdout/stderr of the forked command to your GUI ? I think it will be the easier way to do it.

For example I'm calling Usvfs.run() with artificially generated Args, it runs in a thread, it's still doesn't have output redirected, but I will make it in next revision.

**Comment by sobkas** [comment]
> [@sobkas](https://github.com/sobkas) Not sure. If im not mistaken in the official GUI for gamma in windows it has a "update gamma" function where when there is a new version/fixes you can update into the latest gamma version without having to do a complete reinstall.

I wasn't able to use official GUI, either to complicated to run it or not so compatible with Proton/Wine.
But I think that "Partial Gamma Setup" tab might do it.

**Comment by sobkas** [comment]
Ok, More fixes:
https://github.com/sobkas/gamma-launcher/commit/918e33c7e9162348d0cc196d8193ef92daa2737e

**Comment by sobkas** [comment]
So adding tab to enable/disable mods is ok?

**Comment by Mord3rca** [comment]
> So adding tab to enable/disable mods is ok?

@sobkas Maybe focus on having a usable & integrated GUI first before adding stuff :)

**Comment by sobkas** [comment]
For some reason 7z while installing Anomaly still writes to terminal:

> 7-Zip 24.09 (x64) : Copyright (c) 1999-2024 Igor Pavlov : 2024-11-29
>  64-bit locale=pl_PL.utf8 Threads:32 OPEN_MAX:1024, ASM
> 
> Scanning the drive for archives:
> 1 file, 9742156148 bytes (9291 MiB)        
> 
> Extracting archive: /home/sobkas/gamma_tests/anomaly/Anomaly-1.5.3-Full.2.7z
> --
> Path = /home/sobkas/gamma_tests/anomaly/Anomaly-1.5.3-Full.2.7z
> Type = 7z
> Physical Size = 9742156148
> Headers Size = 2453
> Method = LZMA2:26 LZMA:20 BCJ2
> Solid = +
> Blocks = 2
> 
>  45% 59 - db/meshes/meshes_dynamics.db0

**Comment by sobkas** [comment]
https://github.com/sobkas/gamma-launcher/commit/24e8d9f2301e1ec2c2667256869ea87745604488
Once field input was provided in one tab, all tabs that have same field share same field input for this field

https://github.com/user-attachments/assets/c9ac66d2-ea76-4a79-a3ec-9d4133c57415

**Comment by sobkas** [comment]
Persistent text fields input, so you don't have to enter every directory after GUI restart

**Comment by Mord3rca** [comment]
@sobkas Not really ... Just tell me what your GUI need and I'll take care of pip and pyinstaller :)

**Comment by sobkas** [comment]
> [@sobkas](https://github.com/sobkas) Not really ... Just tell me what your GUI need and I'll take care of pip and pyinstaller :)

I already added PyGObject to pyproject.toml and it should be enough

**Comment by Mord3rca** [comment]
@sobkas Looks OK to me. However I don't plan on using it ^^"
Can you create a pull request for it so I can comment the code / usability of the GUI ?

I don't think technical discussion have a place here.

**Comment by sobkas** [comment]
> [@sobkas](https://github.com/sobkas) Looks OK to me. However I don't plan on using it ^^" Can you create a pull request for it so I can comment the code / usability of the GUI ?
> 
> I don't think technical discussion have a place here.

@Mord3rca I'm not talking about code, I'm asking is GUI looking pretty/useful?

**Comment by Mord3rca** [comment]
Hello, can some of you try the GUI [here](https://github.com/sobkas/gamma-launcher/tree/master) ?

To install it, checkout the code and, in the source directory, type: `pip install '.[gui]'`
Then you can use the GUI with: `gui-gamma-launcher`

**Comment by sobkas** [comment]
On unrelated note, does switching to python-unrardll would be ok?
I'm making debian package and I need to use python-unrardll.
Other unrar module wasn't packaged.

**Comment by sobkas** [comment]
Here a deb for Debian https://sobkas.github.io/gamma-launcher_3.1-6_all.deb
Ultra light, had to swap unrar for unrardll because unrar is not in debian.

**Comment by sobkas** [comment]
> * **Flatpak:** Lets have a GUI merged into master first. The packaging solution will come in a second time.
> 
>     * **unrar/unrardll:** I'm not OK to change once again the unrar python lib used.
>       Also, if the flatpak/snap is done after merging the GUI, it kinda become irrelevant.

I'm not going to change anything, with unrar, I'm talking that if there is no unrar(in case of debian package for example) it will try to use unrardll.

**Comment by clockwinder** [comment]
New gamma compatibility project, for those unaware: https://github.com/FaithBeam/stalker-gamma-launcher-clone
